frontend/src/utils/filterHelpers.js

```diff
@@ -164,15 +164,26 @@ export function populateFiltersCaptions(filters) {
           }
         });
       } else {
-        const originalFilter = filterData.filter(
-          (item) => item.filterId === filterId
-        );
+        let originalFilter;
+
+        filterData.forEach((filter) => {
+          if (filter.filterId === filterId) {
+            originalFilter = filter;
+          } else if (filter.includedFilters) {
+            filter.includedFilters.forEach((included) => {
+              if (included.filterId === filterId) {
+                originalFilter = included;
+              }
+            });
+          }
+        });
+
         captionsArray = [originalFilter.caption, originalFilter.caption];
       }
 
       if (captionsArray.join('').length) {
-        filtersCaptions[filter.filterId] = captionsArray;
         filtersCaptions[filterId] = captionsArray;
+        filtersCaptions[filterIdx] = captionsArray;
```

Comment:
why mix ID keys and Index keys in the same map? what's the reason behind it?

Reply:
I have literally no idea. It was like that before, just used a variable here. There might be some shenanigans when rendering but I haven't written that so did not want to break it.

Reply:
understand